backend/assignments/2/src/main/java/com/caching/service/controller/GeoLocationController.java

```diff
@@ -0,0 +1,95 @@
+package com.caching.service.controller;
+
+import com.caching.service.dto.ResponseDTOForward;
+import com.caching.service.dto.ResponseDTOReverse;
+import com.caching.service.model.ForwardGeoCodingParams;
+import com.caching.service.service.GeoLocationService;
+import com.caching.service.model.ReverseGeoCodingParams;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.io.IOException;
+import java.util.List;
+
+@RestController
+@Slf4j
+public class GeoLocationController {
+       public GeoLocationController(GeoLocationService service, ForwardGeoCodingParams params, ReverseGeoCodingParams reverseGeoCodingParams) {
+        this.service = service;
+        this.params = params;
+        this.reverseGeoCodingParams = reverseGeoCodingParams;
+    }
+
+
+    private GeoLocationService service;
+
+    private ReverseGeoCodingParams reverseGeoCodingParams;
+
+    private ForwardGeoCodingParams params;
+
+    /**
+     * Method to perform forward geocoding based on the given address.
+     *
+     * @param  address   the address for geocoding
+     * @return          a list of response DTOs for forward geocoding
+     */
+    @GetMapping(value = "/geocoding")
+    public List<ResponseDTOForward> forwardGeoCoding(@RequestParam("address") String address) throws IOException {
+        params.setAddress(address);
+        log.info(params.getAddress());
+
+        List<ResponseDTOForward> locations = service.performForwardGeoCoding(params);
+
+        try {
+
+            ResponseDTOForward firstLocation = locations.get(0);
+            log.debug("first location for null check "+firstLocation);
+            return locations;
+
```

Comment:
The controller can pass the `address` to the service. There an initial method can handle all this.

Reply:
Same comment for both APIs